### PR TITLE
Morph table plugin

### PR DESCRIPTION
**Pull request by fbaumgardt**
Feel free to test run this first. There's still a few kinks:

1. morph-form-create has undo/redo but the entry doesn't get removed from localStorage.
2. I'd like to change the layout of the morph-form-create template, but don't know if it conflicts with its other uses.
3. inline-comment submits on `<Enter>` but there's no documentation for it.
4. inline-comment is just a string for now, no validation or destructuring.

We can move these into issues once you've had a chance to look at the plugin.

### CODE REVIEW

**Comment by balmas** [comment]
@fbaumgardt thanks! My morning got away from me but I'm turning to look at this now.

File: app/js/arethusa.inline_comments/directives/inline_comment.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    function(state) {
+        return {
+            restrict: 'A',
+            scope: {
+                token:"=commentToken"
+            },
+            link: function(scope, element, attrs) {
+                scope.comment = scope.token.comment? scope.token.comment : '';
+
+                //TODO: undo function
```
Comment (balmas) [inline app/js/arethusa.inline_comments/directives/inline_comment.js]: maybe we can remove these comments? it seems like this is addressed now?

**Comment by balmas** [comment]
I think this is really great. The code itself is hard for me to find fault with. Most of my comments are for pretty small improvements.   The first three I think should be done now:

1 - I think we should use the translate service to populate the table headings ... this is simple you just have to replace the inline text in the template with a translate attribute, e.g.

``` <th translate="table.ID"/>```

and then add the strings to the en.json (and at least placeholders in the other language files there, e.g
```
...
,
  "table" : {
    "ID": "ID"
  }
```

2 - I think the inline comments should submit when a user leaves the text box, rather than on enter , using ng-blur rather than ng-change. This avoid the issue of needing to document the behavior on enter and I think it will be more what the user would expect. We could check to see if they changed any text before acting on the event.

3 - I'd like to see unit tests added for the new services and directives.  I think we should also add a test for the retrieving/persisting of the comment field to the treebank retriever/persister tests.

The next changes are enhancements I think we could add at a later time:

4 - re the redo/undo on the morph-form-create: this is handled a different way in the morph3 template -- the expandSelection setting is checked, then you get an additional component on the page, the morph-form-attributes directive, which offers a remove form option (along with create and report error, the latter of which is meaningless at the moment.) I don't think we want to include this directive as-is on the table display, but we could either add something to activate a removeForm function, or we could add redo/undo to the morph-form-create directive.  Probably the latter would be better, but it might need some thought. This might be good to do in conjunction with the changes you want to make to the create-morph-form template.

5 - it might be nice to have some of the other plugins available in the config with the table -- I could see at least wanting the history display so that undo/redo are less opaque. search might be very useful too.

**Comment by fbaumgardt** [comment]
Thanks for the pointers in 1,2,4!

3 is next on the list (will remove the TODO then) and wrt 5 - the conf was just a distraction-free dev environment and i should actually run some different layouts to test.

We can also debate either moving inline-comments into the table plugin or fleshing it out into a more functional thing.

**Comment by balmas** [comment]
Sounds good. I would like to put this up in the staging environment to get some feedback from Timo and Giuseppe as soon as you're ready. I can merge now or wait until items 1-3 are done. whichever you prefer!

**Comment by fbaumgardt** [comment]
Given that 1&2 are small and I'm already on 3, let's finish those first.

I might have a question tomorrow about the testing interface, if I don't find the answer myself before then.

**Comment by fbaumgardt** [comment]
The new commits cover 1-3. The tests can certainly be improved, but I'd be happy to let Giuseppe and Timo take a look at it now.

**Pull request merged**